cli/cli.ts

```diff
@@ -2591,6 +2591,113 @@ function testDirAsync(dir: string) {
         })
 }
 
+function testDecompilerAsync(dir: string): Promise<void> {
+    const filenames: string[] = [];
+
+    const baselineDir = path.resolve(dir, "baselines")
+
+    try {
+        const stats = fs.statSync(baselineDir);
+        if (!stats.isDirectory()) {
+            return Promise.reject(baselineDir + " is not a directory");
+        }
+    }
+    catch (e) {
+        return Promise.reject(baselineDir + " does not exist");
+    }
+
+    for (const file of fs.readdirSync(dir)) {
+        if (file[0] == ".") {
+            continue;
+        }
+
+        const filename = path.join(dir, file)
+        if (U.endsWith(file, ".ts")) {
+            filenames.push(filename)
+        }
+    }
+
+    return Promise.mapSeries(filenames, filename => {
+        const basename = path.basename(filename);
+        const baselineFile = path.join(baselineDir, replaceFileExtension(basename, ".blocks"))
+
+        let baselineExists: boolean;
+        try {
+            const stats = fs.statSync(baselineFile)
+            baselineExists = stats.isFile()
+        }
+        catch (e) {
+            baselineExists = false
+        }
+
+        if (!baselineExists) {
+            // Don't kill the promise chain, just print an resolve
+            console.log(`Could not find baseline file for "${basename}" at ${baselineFile}`)
+            return Promise.resolve()
+        }
+
+        return decompileAsyncWorker(filename)
+            .then(decompiled => {
+                const baseline = fs.readFileSync(baselineFile, "utf8")
+                if (compareBaselines(decompiled, baseline)) {
+                    console.log(`${basename} - okay`);
+                }
+                else {
+                    const outFile = path.join(replaceFileExtension(filename, ".local.blocks"))
+                    fs.writeFileSync(outFile, decompiled)
+                    console.log(`${basename} - error: baselines did not match (local baseline written to ${outFile})`);
+                }
+            }, error => {
+                console.log(`${basename} - error: ${error}`)
+            })
+    })
+        .then(() => {});
+}
+
+function compareBaselines(a: string, b: string): boolean {
+    // Ignore whitespace
+    return a.replace(/\s/g, "") === b.replace(/\s/g, "")
+}
+
+function replaceFileExtension(file: string, extension: string) {
+    return file && file.substr(0, file.length - path.extname(file).length) + extension;
+}
+
+function decompileAsync(...fileNames: string[]) {
+    return Promise.mapSeries(fileNames, f => {
+        const outFile = replaceFileExtension(f, ".blocks")
+        return decompileAsyncWorker(f)
+            .then(result => {
+                fs.writeFileSync(outFile, result)
+                console.log("Wrote " + outFile)
+            })
+    })
+        .then(() => {
+            console.log("Done")
+        }, error => {
+            console.log("Error: " + error)
```

Comment:
if we want to automate the test suite, an error should fail the execution of the CLI